code/modules/power/groundmap_geothermal.dm

```diff
@@ -357,7 +352,239 @@ GLOBAL_VAR_INIT(generators_on_ground, 0)
 	name = "\improper Reinforced Reactor Turbine"
 	is_corruptible = FALSE
 
-#undef GEOTHERMAL_NO_DAMAGE
-#undef GEOTHERMAL_LIGHT_DAMAGE
-#undef GEOTHERMAL_MEDIUM_DAMAGE
-#undef GEOTHERMAL_HEAVY_DAMAGE
+
+/*	Thermo-bluespace generator -- the main, big boy generator that engineers need to be concerned with
+	While active -- Generates a shitton of power and overclocks disk generation speed
+	Xenos can disable it after a short windup, which if the generator is past 50%, triggers a massive explosion */
+/obj/machinery/power/geothermal/tbg
+	name = "\improper Thermo-Bluespace Generator"
+	desc = "A marvel of modern engineering and a shining example of pioneering bluespace technology, able to power entire colonies with very little material consumption - perfectly suited for isolated areas on the outer rim.\nHighly volatile, but that shouldn't matter on some quiet backwater colony, right..?"
+	icon = 'icons/obj/machines/tbg.dmi'
+	power_generation_max = 5000000 //Powers an entire colony
+	time_to_break = 5 SECONDS
+	voice_filter = "alimiter=0.9,acompressor=threshold=0.2:ratio=20:attack=10:release=50:makeup=2,highpass=f=1000"
```

Comment:
whered you get this out of curiosity

Reply:
Cigarette vendors, roomba, etc all use it